bazel/BUILD

```diff
@@ -1,25 +1,76 @@
 load("@bazel_skylib//lib:selects.bzl", "selects")
 
 config_setting(
-    name = "runtime_v8",
+    name = "requested_engine_v8",
+    values = {"define": "engine=v8"},
+)
+
+config_setting(
+    name = "requested_runtime_v8",
     values = {"define": "runtime=v8"},
 )
 
+selects.config_setting_group(
+    name = "engine_v8",
+    match_any = [
+        ":requested_engine_v8",
+        ":requested_runtime_v8",
+    ],
+)
+config_setting(
+    name = "requested_engine_wamr",
+    values = {"define": "engine=wamr"},
+)
```

Comment:
We don't have any public documentation, nor any downstream consumers other than Envoy, so I'm not sure if that level of backward compatibility is necessary. What do you think, @mathetake?

Reply:
I don't think we need this level of backward compatibility, so I don't mind removing `runtime=*`options

Reply:
Done (along with mass rename in the codebase).